crates/wasmtime/src/runtime/vm.rs

```diff
@@ -202,6 +202,32 @@ pub unsafe trait VMStore {
     /// as a trap to clean up Wasm execution.
     fn maybe_async_gc(&mut self, root: Option<VMGcRef>) -> Result<Option<VMGcRef>>;
 
+    /// Callback invoked whenever an instance needs to grow the GC heap.
+    ///
+    /// Cooperative, async-yielding (if configured) is completely transparent.
+    ///
+    /// If the growth was canceled or fails, returns an error.
+    fn maybe_async_grow_gc_heap(&mut self, bytes_needed: u64) -> Result<()>;
```

Comment:
Would it be possible to avoid the split between either GC or grow? My hope would be that there's a single location which makes a determination of what to do. I might just not be familiar with the code and the method below is The Location, but I feel like the logic of exactly whether to grow or GC is spread out across multiple files at this time.

I understand we probably want a method-per-op at some point but  I would hope that would be buried in some guaranteed-module-private location rather than so prominently exposed to the entire codebase through the `VMStore` trait.